src/main/java/mif/vu/ikeea/RepositoryService/RoleService.java

```diff
@@ -0,0 +1,49 @@
+package mif.vu.ikeea.RepositoryService;
+
+import mif.vu.ikeea.Checker.RoleChecker;
+import mif.vu.ikeea.Entity.ApplicationUser;
+import mif.vu.ikeea.Entity.Repository.RoleRepository;
+import mif.vu.ikeea.Entity.Role;
+import mif.vu.ikeea.Enums.ERole;
+import mif.vu.ikeea.Exceptions.ResourceNotFoundException;
+import mif.vu.ikeea.Payload.UpdateForLeaderRequest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.HashSet;
+import java.util.Set;
+
+@Service
+public class RoleService {
```

Comment:
RepositoryService visos klases yra skirtos tik dirbti su DB! Papildomai jokiu operaciju cia neatliekame - pafetchinam is DB + metame exception jei kazkas negerai. Viksas. Updatint roles ir jas keisti reikia kitos klases - pvz. Promotion... kazkas, na sugalvok pavadinima koki :)